src/FGame.cpp

```diff
@@ -18,18 +18,27 @@ FGame::FGame(std::shared_ptr<FEngine> engine)
 	m_window = std::unique_ptr<sf::RenderWindow>(window);		
 }
 
-// TODO: Find a way to keep textures not destroyed
-//sf::Sprite FGame::loadSprite(std::string fileName)
-//{
-//	sf::Texture texture;
-//	if (!texture.loadFromFile(fileName, sf::IntRect(0, 0, m_tileSize, m_tileSize)))
-//	{
-//		std::cout << "ERROR in texture loading at " << __LINE__ << " in File" << __FILE__ << std::endl;
-//	}
-//	sf::Sprite sprite;
-//	sprite.setTexture(texture);
-//	return sprite;
-//}
+std::shared_ptr<sf::Sprite> FGame::loadSprite(std::string fileName, int width, int height)
+{
+	auto exists = m_sprites.find(fileName);
+	if (exists == m_sprites.end())
+	{
+		auto texture = std::make_shared<sf::Texture>();
+		if (!texture->loadFromFile(fileName, sf::IntRect(0, 0, width, height)))
+		{
+			std::cout << "ERROR trying to load " << fileName;
+		}
+		auto sprite = std::make_shared<sf::Sprite>();
+		sprite->setTexture(*texture);
+		m_sprites[fileName] = sprite;
+		m_textures[fileName] = texture;
+		return sprite;
```

Comment:
Yes, for now it assumes 1-1 relationship, in future we could improve it by using 1 texture for many sprites.